modules/dnn/misc/python/test/test_dnn.py

```diff
@@ -546,5 +546,21 @@ def test_net_builder(self):
         out = net.forward()
         self.assertEqual(out.shape, (1, 2, 3, 4))
 
+    def test_bool_operator(self):
+        n = self.find_dnn_file('dnn/onnx/models/and_op.onnx')
+        x = self.find_dnn_file('dnn/onnx/data/input_and_op_0.npy')
```

Comment:
Can we generate test data in runtime instead?

Reply:
Technically yes, but how to do that without adding extra funtionality in `modules/dnn/misc/python/test/test_dnn.py` file?

Reply:
How would you implement this functionality?

Reply:
There is no generic test in test_dnn.py - you can create input/output in runtime as numpy array

Reply:
Alternatively, let's make input data more complex - initialize them randomly instead of zeros/ones

https://github.com/opencv/opencv_extra/pull/1201/files#diff-066596aee5ba198740fae15d7e39dd3c0bebcec5c6d5faab96a61c2c5de97863R404